pydatastructs/linear_data_structures/arrays.py

```diff
@@ -255,6 +258,9 @@ def delete(self, idx):
     def size(self):
         return self._size
 
+    def __str__(self):
+        return str(self._data[:self._last_pos_filled+1])
```

Comment:
Can we shift the following code here, 
https://github.com/codezonediitj/pydatastructs/blob/6605b9761a920632a69e34547765a2c2b78cbb13/pydatastructs/trees/binary_trees.py#L150-L156

And modify the `str` calls wherever, `DODA` is used.

Reply:
If I use `DODA.__str__()` it wont print the key of each node it would print something like this
`[<pydatastructs.utils.misc_util.TreeNode object at 0x000001F45C54E048>]` so we still have to override the str method in binary_trees.py

Reply:
```python
def __str__(self): 
     to_be_printed = ['' for i in range(self._last_pos_filled + 1)] 
     for i in range(self._last_pos_filled + 1): 
         if self._data[i] is not None:  
             to_be_printed[i] = str(self._data[i])
     return str(to_be_printed) 
```
The above may break tests for trees if their `str` method is removed. We can keep both the methods(`BinaryTree.__str__` and `DODA.__str__`) as it's just about printing things, and they both will look similar.

Reply:
ok I get it.

Reply:
here is the thing, if I take str of each element, the output would be `"['1', '2', '3', '4]"` instead of `'[1, 2, 3, 4]'`

Reply:
Let it be.

Reply:
Done.